### ISSUE TITLE
Invalid distribution-name value Error

### ISSUE THREAD

**Issue by ndynamite** [tidev/titanium#595]
Hi,

When building for a device or for the simulator I now get the error
`[ERROR] Invalid "--distribution-name" value "Apple Distribution: XXXXX XXX XXX (XXXXXXXXXX)"`

I have replaced the name of the certificate with Xs here.

This happens when running ti build and including the '--distribution-name'.

If I just run ti build with a target, and don't include the distribution-name it then prompts to 'Select a certificate by number or name' which works. 

This isn't a big issue as there is an easy work around, I just thought I should log it here. 

Thanks.

**Comment by hbugdoll** [comment]
In general the exact command line build command would have been helpful.

The error message `[ERROR] Invalid "--distribution-name" value "..."` points to https://github.com/tidev/titanium-cli/blob/d0c8a86871b25da1b7d5be013fc89e2c7f609fbf/src/cli.js#L1480
in `validate()`. Ah, now I see, it's only raised if prompting is disabled (`--no-prompt`).

When prompting is enabled the error changes to the more clear message
`[ERROR] Invalid distribution certificate "..."` from the SDK itself
https://github.com/tidev/titanium-sdk/blob/8378d3b5c7ae2f4970501eb4994319deaaafb7d5/iphone/cli/commands/_build.js#L875

@ndynamite That means that your specified distribution certificate is not valid for this build case (e.g. it has the wrong App ID).
I definitely believe that the certificate specified via `--distribution-name` is not included in the prompted selection list.

----

Use of `--distribution-name` checked with Ti CLI 7 & 8 without any problems.
In my opinion this issue can be closed.

**Comment by cb1kenobi** [comment]
Yeah, the prompting route is handled in the distribution name's option handler in the SDK: https://github.com/tidev/titanium-sdk/blob/93d2af6d606dbbc365d472a408e7f4e8b4804a58/iphone/cli/commands/_build.js#L855.

If building from VS Code, I believe prompting is disabled and thus you get the generic message.

In any case, I'll close this. Thanks!
